src/LoginWithAmazon.test.js

```diff
@@ -1,33 +1,53 @@
 import React from 'react';
-import TestUtils from 'react';
-import { shallow, mount } from 'enzyme';
+import {shallow, mount} from 'enzyme';
 import LoginWithAmazon from './LoginWithAmazon';
 
 let loginWithAmazon;
 let loginWithAmazonInstance;
-let mockSetAccessToken;
-let authorize;
+let mockSetAuthorization;
+let amazonAuthorizationSpy;
 
 beforeEach(() => {
-    mockSetAccessToken = jest.fn();
-    loginWithAmazon = shallow(<LoginWithAmazon setAccessToken={mockSetAccessToken}/>);
+    mockSetAuthorization = jest.fn();
+    loginWithAmazon = mount(<LoginWithAmazon setAuthorization={mockSetAuthorization}/>);
     loginWithAmazonInstance = loginWithAmazon.instance();
-    authorize = jest.fn();
+    amazonAuthorizationSpy = jest.fn();
 
     Object.defineProperty(window, 'amazon', {
-        value: { Login: { authorize: authorize } },
-        configurable: true
+        value: { Login: { authorize: amazonAuthorizationSpy } },
+        writable: true
     });
 });
 
+afterEach(() => {
+    loginWithAmazon.unmount();
+});
+
 it('renders without crashing', () => {
-    loginWithAmazon.find('button').simulate('click');
+    const wrapper = shallow(<LoginWithAmazon />);
+    wrapper.unmount();
+});
+
+it('verify amazon authorization called when login button is clicked', () => {
+    mount(<LoginWithAmazon />).find('button').simulate('click');
 
     // Verify authorize called once
-    expect(authorize.mock.calls.length).toBe(1);
+    expect(amazonAuthorizationSpy.mock.calls.length).toBe(1);
+});
+
+it('logs error message when response is undefined', () => {
+    global.console = {
+      log: jest.fn()
+    }
+
+    let response;
```

Comment:
const?

Reply:
const in javascript needs to be initialized during declaration. Since the usecase here needs the variable to be 'undefined', using let instead.